internal/server/storage/drivers/driver_nfs.go

```diff
@@ -0,0 +1,125 @@
+package drivers
+
+import (
+	"strings"
+
+	"github.com/lxc/incus/v6/internal/linux"
+	"github.com/lxc/incus/v6/internal/migration"
+	deviceConfig "github.com/lxc/incus/v6/internal/server/device/config"
+	localMigration "github.com/lxc/incus/v6/internal/server/migration"
+	"github.com/lxc/incus/v6/shared/util"
+)
+
+type nfs struct {
+	dir
+}
+
+// Info returns info about the driver and its environment.
+func (n *nfs) Info() Info {
+	return Info{
+		Name:                         "nfs",
+		Version:                      "1",
+		DefaultVMBlockFilesystemSize: deviceConfig.DefaultVMBlockFilesystemSize,
+		OptimizedImages:              false,
+		PreservesInodes:              false,
+		Remote:                       n.isRemote(),
+		VolumeTypes:                  []VolumeType{VolumeTypeBucket, VolumeTypeCustom, VolumeTypeImage, VolumeTypeContainer, VolumeTypeVM},
```

Comment:
So initially my thought was to only allow `VolumeTypeCustom` on there, so only allowing NFS to be used for additional storage to instances either stored on local storage or stored on a proper block shared storage solution (clustered LVM, Ceph or Linstor).

I think we should at least exclude `VolumeTypeBucket` and `VolumeTypeImage`.

Container and VM, we'd need to see how well they work. My gut feeling given NFS locking is "not well", the question is then whether we still allow it but document that it's really not recommended or we plain not allow it and only keep custom volumes on it.

Reply:
Well it depends on the amount of work required. 4 hours of work that could potentially go to the trash is ok. 4 days, not so much.